lsp/nls/src/files.rs

```diff
@@ -140,8 +144,8 @@ fn associate_failed_import(server: &mut Server, err: &nickel_lang_core::error::E
 
 pub(crate) fn typecheck(
     server: &mut Server,
-    file_id: FileId,
-) -> Result<CacheOp<()>, Vec<Diagnostic<FileId>>> {
+    file_id: CacheKey,
+) -> Result<(), Vec<Diagnostic<CacheKey>>> {
```

Comment:
I vaguely remember `CacheOp<()>` being a useful distinction from `()` to avoid cycles/loops when typechecking or importing circular stuff. But I'm not sure.

Reply:
Grepping through the source code, by now `CacheOp` is used in a single place in the LSP (apart from the cache itself) and that is an extension function to the cache. So I think it's safe to get rid of it in the external interface of the cache.